lib/whitespace.coffee

```diff
@@ -77,14 +77,42 @@ class Whitespace
 
   convertTabsToSpaces: (editor) ->
     buffer = editor.getBuffer()
-    spacesText = new Array(editor.getTabLength() + 1).join(' ')
+
+    tabLength = editor.getTabLength()
 
     buffer.transact ->
-      buffer.scan /\t/g, ({replace}) -> replace(spacesText)
+      buffer.scan /^.*\t.*$/g, ({matchText, replace}) ->
```

Comment:
Seems like you could continue to scan for single `\t` characters and use the associated range to determine how many spaces to replace this with.